4-Beyond-Classical-Search/c_hillClimbing.js

```diff
@@ -1,144 +1,166 @@
-        this.hillData[maximas[i]].maxima = true;
-      }
 
+    this.xScale = d3.scaleLinear()
+      .domain([0, this.states.length])
+      .range([this.padding, this.w - this.padding]);
+    this.blockWidth = (this.xScale(this.states.length) - this.xScale(0)) / this.states.length;
+    this.yScale = d3.scaleLinear()
+      .domain([0, d3.max(this.states)])
+      .range([this.padding, this.h - this.padding]);
+    this.renderHill();
 
-      this.xScale = d3.scaleLinear()
-        .domain([0, this.states.length])
-        .range([this.padding, this.w - this.padding]);
-      this.blockWidth = (this.xScale(this.states.length) - this.xScale(0)) / this.states.length;
-      this.yScale = d3.scaleLinear()
-        .domain([0, d3.max(this.states)])
-        .range([this.padding, this.h - this.padding]);
-      this.renderHill();
+  }
 
-    }
-    renderHill() {
-      this.svgHill = this.svg.selectAll('.block')
-        .data(this.hillData)
-        .enter()
-        .append('g')
-        .attr('class', 'block');
-      this.svgRects = this.svgHill
-        .append('rect')
-        .attr('height', d => this.yScale(d.objective))
-        .attr('width', this.blockWidth)
-        .attr('x', (d) => this.xScale(d.state))
-        .attr('y', (d) => this.h - this.yScale(d.objective))
-        .style('opacity', (d) => (d.visited) ? 1 : 0)
-        .style('fill', 'hsl(217,61.2%,53.5%)')
-        .style('border', '1px solid');
-    }
+  renderHill() {
+    this.svgHill = this.svg.selectAll('.block')
+      .data(this.hillData)
+      .enter()
+      .append('g')
+      .attr('class', 'block');
+    this.svgRects = this.svgHill
+      .append('rect')
+      .attr('height', d => this.yScale(d.objective))
+      .attr('width', this.blockWidth)
+      .attr('x', (d) => this.xScale(d.state))
+      .attr('y', (d) => this.h - this.yScale(d.objective))
+      .style('opacity', (d) => (d.visited) ? 1 : 0)
+      .style('fill', colors.hill)
+      .style('border', '1px solid');
+  }
 
-    visit(state) {
-      this.hillData[state].visited = true;
-      this.svgRects.filter((d) => d.state == state)
-        .transition()
-        .duration(100)
-        .style('opacity', 1);
-    }
+  visit(state) {
+    this.hillData[state].visited = true;
+    this.svgRects.filter((d) => d.state == state)
+      .transition()
+      .duration(100)
+      .style('opacity', 1);
+  }
 
-    showAll() {
-      this.svgRects.transition()
-        .duration(100)
-        .style('opacity', (d) => {
-          return (d.visited) ? 1 : 0.6;
-        })
-        .style('fill', (d) => {
-          if (d.maxima) {
-            if (d.visited) {
-              return 'hsl(110, 100%, 38%)';
-            } else {
-              return 'hsl(102, 100%, 56%)';
-            }
+  showAll() {
+    this.svgRects.transition()
+      .duration(100)
+      .style('opacity', (d) => {
+        return (d.visited) ? 1 : 0.6;
+      })
+      .style('fill', (d) => {
```

Comment:
Color implementation suggestion:

Right now you have some colors defined in a color object and other colors defined in the page CSS. 

Instead of setting the fill color here, use `.attr('class', 'visited')` or `.classed('visited', true)`, and then you'll be able to set all the fill colors in the page CSS, in one place. That also will make it easier to override the colors for one diagram while reusing code across diagrams.